### PR TITLE
07 memory management prog

### PR DESCRIPTION
**Pull request by DomninDmitry-dev**
Homework MemoryManagement test program.
This program testing:
- using functions: malloc, calloc, alloca;
- measure time of each allocation/freeing.

Build this program:
make SYSTEM=qemu
Clean this program:
make clean

### CODE REVIEW

File: 07-MemoryManagement/user_space/myprog.c [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <time.h>
+#include <sys/time.h>
+#include <math.h>
+
+int i, *p;
```
Comment (yekovalyov) [inline 07-MemoryManagement/user_space/myprog.c]: What is the reason to make i and p static variable.

File: 07-MemoryManagement/user_space/myprog.c [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+long before, relalloc, relfree;
+long allocsize[5];
+
+int main(void)
+{
+	allocsize[0] = (long)pow(2, 10);
+	allocsize[1] = (long)pow(2, 15);
+	allocsize[2] = (long)pow(2, 22);
+	allocsize[3] = (long)pow(2, 32);
+	allocsize[4] = (long)pow(2, 35);
```
Comment (yekovalyov) [inline 07-MemoryManagement/user_space/myprog.c]: No all required buffers are allocated.

File: 07-MemoryManagement/user_space/myprog.c [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+			printf("%20ld\t", (long)allocsize[i]);
+			printf("Malloc error\n");
+			continue;
+		}
+		clock_gettime(CLOCK_MONOTONIC, &ts);
+		relalloc = ts.tv_nsec - before;
+		before = ts.tv_nsec;
+		free(p);
+		clock_gettime(CLOCK_MONOTONIC, &ts);
+		relfree = ts.tv_nsec - before;
```
Comment (yekovalyov) [inline 07-MemoryManagement/user_space/myprog.c]: Is this calculation of time duration correct?

File: 07-MemoryManagement/user_space/myprog.c (hunk 2 above)
Comment (DomninDmitry-dev) [reply 07-MemoryManagement/user_space/myprog.c]: What do you mean?

Comment (DomninDmitry-dev) [reply 07-MemoryManagement/user_space/myprog.c]: I understood, it was necessary from 0 to 64.

File: 07-MemoryManagement/user_space/myprog.c (hunk 3 above)
Comment (DomninDmitry-dev) [reply 07-MemoryManagement/user_space/myprog.c]: I think yes, because we use a high resolution timer.

Comment (yekovalyov) [reply 07-MemoryManagement/user_space/myprog.c]: Is it possible to get negative values in relfree?

Comment (DomninDmitry-dev) [reply 07-MemoryManagement/user_space/myprog.c]: Yes maybe. Now I take seconds.

**Comment by DomninDmitry-dev** [comment]
Optimized program code.

**Pull request merged**